### PR TITLE
 HistoryDB: Create table with archs, use them as foreign keys in rpm table

### PR DESCRIPTION
**Pull request by jrohel**
Please review and merge firstly PR https://github.com/rpm-software-management/dnf5/pull/34
Then I will rebase this PR.

### CODE REVIEW

File: libdnf/transaction/db/archs.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#ifndef LIBDNF_TRANSACTION_DB_ARCHS_HPP
+#define LIBDNF_TRANSACTION_DB_ARCHS_HPP
+
+#include "utils/sqlite3/sqlite3.hpp"
+
+#include <memory>
+
+namespace libdnf::transaction {
+
+/// Create a query (statement) that inserts new records to the 'pkg_names' table
```
Comment (jan-kolarik) [inline libdnf/transaction/db/archs.hpp:29]: Please update comment with the correct table name.

File: libdnf/transaction/db/archs.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+std::unique_ptr<utils::SQLite3::Statement> archs_insert_if_not_exists_new_query(utils::SQLite3 & conn);
+
+/// Use a query to insert a new record to the 'pkg_names' table
```
Comment (jan-kolarik) [inline libdnf/transaction/db/archs.hpp:32]: Same as above...

**Review by jan-kolarik — COMMENTED** [review]
LGTM, just two mentioned "typos".

File: libdnf/transaction/db/archs.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+static constexpr const char * SQL_GET_ID = R"**(
+    SELECT
+        id
+    FROM
+        archs
+    WHERE
+        name = ?
+)**";
+
+std::unique_ptr<utils::SQLite3::Statement> archs_get_id_new_query(utils::SQLite3 & conn) {
```
Comment (jan-kolarik) [inline libdnf/transaction/db/archs.cpp:59]: Unused, probably to be deleted? Or reference in `hpp`?

File: libdnf/transaction/db/archs.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    return std::make_unique<utils::SQLite3::Statement>(conn, SQL_GET_ID);
+}
+
+int64_t archs_get_id(utils::SQLite3::Statement & query, const std::string & name) {
```
Comment (jan-kolarik) [inline libdnf/transaction/db/archs.cpp:63]: Same as above...

**Review by jan-kolarik — COMMENTED** [review]
I've also noticed some unused code.

File: libdnf/transaction/db/archs.cpp (hunk 3 above)
Comment (jrohel) [reply libdnf/transaction/db/archs.cpp]: It's a remnant. I was thinking about the optimal solution for working with tables. I'll delete, internal optimizations can be solved later.

File: libdnf/transaction/db/archs.cpp (hunk 4 above)
Comment (jrohel) [reply libdnf/transaction/db/archs.cpp]: It's a remnant. I was thinking about the optimal solution for working with tables. I'll delete, internal optimizations can be solved later.

**Pull request merged**